test/prob/test_fixture_ccdf_log.hpp

```diff
@@ -582,124 +563,170 @@ class AgradCcdfLogTestFixture : public ::testing::Test {
-        test_multiple_gradient_values(
-            is_vector<T3>::value, single_ccdf_log, single_gradients2,
-            pos_single, multiple_gradients2, pos_multiple, N_REPEAT);
-      if (!is_constant_all<T4>::value && !is_empty<T4>::value
-          && (std::is_same<Scalar4, fvar<var>>::value
-              || std::is_same<Scalar4, fvar<fvar<var>>>::value))
-        test_multiple_gradient_values(
-            is_vector<T4>::value, single_ccdf_log, single_gradients2,
-            pos_single, multiple_gradients2, pos_multiple, N_REPEAT);
-      if (!is_constant_all<T5>::value && !is_empty<T5>::value
-          && (std::is_same<Scalar5, fvar<var>>::value
-              || std::is_same<Scalar5, fvar<fvar<var>>>::value))
-        test_multiple_gradient_values(
-            is_vector<T5>::value, single_ccdf_log, single_gradients2,
-            pos_single, multiple_gradients2, pos_multiple, N_REPEAT);
-
-      pos_single = 0;
-      pos_multiple = 0;
-      if (!is_constant_all<T0>::value && !is_empty<T0>::value
-          && std::is_same<Scalar0, fvar<fvar<var>>>::value)
-        test_multiple_gradient_values(
-            is_vector<T0>::value, single_ccdf_log, single_gradients3,
-            pos_single, multiple_gradients3, pos_multiple, N_REPEAT);
-      if (!is_constant_all<T1>::value && !is_empty<T1>::value
-          && std::is_same<Scalar1, fvar<fvar<var>>>::value)
-        test_multiple_gradient_values(
-            is_vector<T1>::value, single_ccdf_log, single_gradients3,
-            pos_single, multiple_gradients3, pos_multiple, N_REPEAT);
-      if (!is_constant_all<T2>::value && !is_empty<T2>::value
-          && std::is_same<Scalar2, fvar<fvar<var>>>::value)
-        test_multiple_gradient_values(
-            is_vector<T2>::value, single_ccdf_log, single_gradients3,
-            pos_single, multiple_gradients3, pos_multiple, N_REPEAT);
-      if (!is_constant_all<T3>::value && !is_empty<T3>::value
-          && std::is_same<Scalar3, fvar<fvar<var>>>::value)
-        test_multiple_gradient_values(
-            is_vector<T3>::value, single_ccdf_log, single_gradients3,
-            pos_single, multiple_gradients3, pos_multiple, N_REPEAT);
-      if (!is_constant_all<T4>::value && !is_empty<T4>::value
-          && std::is_same<Scalar4, fvar<fvar<var>>>::value)
-        test_multiple_gradient_values(
-            is_vector<T4>::value, single_ccdf_log, single_gradients3,
-            pos_single, multiple_gradients3, pos_multiple, N_REPEAT);
-      if (!is_constant_all<T5>::value && !is_empty<T5>::value
-          && std::is_same<Scalar5, fvar<fvar<var>>>::value)
-        test_multiple_gradient_values(
-            is_vector<T5>::value, single_ccdf_log, single_gradients3,
-            pos_single, multiple_gradients3, pos_multiple, N_REPEAT);
+      EXPECT_EQ(single_gradients1.size(), multiple_gradients1.size())
+          << "scalar and vectorized results should produce the same number of "
+             "first order gradients";
```

Comment:
Hmm, explain more. I'm pretty sure this is making sure the relationship:

```
lccdf(a) + lccdf(b) + lccdf(c) + ... = lccdf([a, b, c, ...])
```

holds. What do you mean by repeat scalar gradients?

> Also I think this can't fail due to a bug in the tested distribution

Well it definitely failed some of the tests to get this greenlit lol. I do need to check though that this actually would have prevented the original problem.

Reply:
If I understand correctly what you are doing in this test is (d means derivative and vec variables in the following are scalars, this just means they are used in vector implementation):
`lccdf(a_scalar) + lccdf(b_scalar) + lccdf(c_scalar)`
and
`lccdf([a_vec, b_vec, c_vec, ...])`

And this expect is checking that:
`{d_a_scalar, d_b_scalar, d_c_scalar}.size() == {d_a_vec, d_b_vec, d_c_vec}.size()`, which is obviously true, regardless of what `lccdf()` does.

Reply:
Oh right. I guess I just added these checks to sanity check the function is actually doing the right thing. I wouldn't leave them if this test framework was tested but since it isn't I kinda want to just leave them as a sanity thing.

Reply:
Don't need this check with expect_near_real.